src/libraries/System.Private.Xml/src/System/Xml/XPath/Internal/XPathScanner.cs

```diff
@@ -104,9 +108,19 @@ public bool CanBeFunction
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private void SkipSpace()
         {
-            while (XmlCharType.IsWhiteSpace(CurrentChar) && NextChar()) ;
+            if (XmlCharType.IsWhiteSpace(CurrentChar))
+            {
+                SkipKnownSpace();
+            }
+        }
+
+        private void SkipKnownSpace()
+        {
+            Debug.Assert(XmlCharType.IsWhiteSpace(CurrentChar));
+            while (NextChar() && XmlCharType.IsWhiteSpace(CurrentChar));
```

Comment:
is this faster than what it was before?

Reply:
This loop isn't faster.  What is faster is separating out the non-inlineable loop into its own method, such that the calling SkipSpace (which is on a fast path) is now inlineable. Whitespace is relatively uncommon, so we want the check to skip it to be as fast as possible.